### PR TITLE
Add keygen implement

### PR DESCRIPTION
**Pull request by Tlaster**

### CODE REVIEW

File: ntge/src/bin/ntge-keygen/main.rs [hunk 1]
```diff
@@ -1,4 +1,16 @@
+use dirs::home_dir;
 use gumdrop::Options;
+use ntge_core::*;
+use std::env;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+const DEFAULT_SAVE_PATH: &str = ".cube";
```
Comment (yisiliu) [inline ntge/src/bin/ntge-keygen/main.rs]: `.cube` -> `.ntge`

File: ntge/src/bin/ntge-keygen/main.rs [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+const DEFAULT_FILE_NAME: &str = "id_tessercube";
```
Comment (yisiliu) [inline ntge/src/bin/ntge-keygen/main.rs]: `id_tessercube` -> `id_ntge`

File: ntge/src/bin/ntge-keygen/main.rs [hunk 3]
```diff
@@ -8,15 +20,97 @@ struct NtgeOptions {
     #[options(help = "Print version info and exit.", short = "V")]
     version: bool,
 
-    #[options(help = "Write the result to the file at path OUTPUT. Defaults to standard output.")]
-    output: Option<String>,
+    #[options(help = "Key's file name, default to id_tessercube")]
+    filename: Option<String>,
+
+    #[options(help = "Save path, default to ~/.cube/")]
```
Comment (yisiliu) [inline ntge/src/bin/ntge-keygen/main.rs]: `~/.cube/` -> `~/.ntge/`

**Review by yisiliu — CHANGES_REQUESTED** [review]

**Review by yisiliu — APPROVED** [review]

File: ntge/src/bin/ntge-keygen/main.rs [hunk 4]
```diff
@@ -8,15 +20,87 @@ struct NtgeOptions {
...
+    }
+    let public_key_file = output_file.with_extension(DEFAULT_PUBLIC_KEY_SUFFIX);
+    let private_key_file = output_file.with_extension(DEFAULT_PRIVATE_KEY_SUFFIX);
+    if let Err(_) = create_file_and_write(public_key_file, public_key_content) {
+        return;
+    }
+    if let Err(_) = create_file_and_write(private_key_file, private_key_content) {
+        return;
+    }
+    println!("Successfully create ntge key at {}", output_file.display());
```
Comment (MainasuK) [inline ntge/src/bin/ntge-keygen/main.rs:105]: Basically the Linux/Unix program silence after running.
It's could combine pipe operation If the user chooses printing to stdout.

**Review by MainasuK — APPROVED** [review]

**Pull request merged**